components/Search.tsx

```diff
@@ -1,14 +1,66 @@
-import {FormControl} from "react-bootstrap";
-
-export default function Search() {
-    return (
-        <>
-            <FormControl
-                placeholder="Search notes"
-                autoFocus/>
-            <div className="m-5 text-center text-muted">
-                <p>No recent searches</p>
-            </div>
-        </>
-    );
-}
\ No newline at end of file
+import { FormControl } from "react-bootstrap";
+import useSWR from "swr";
+import { Note } from "@prisma/client";
+import { ListGroup } from "react-bootstrap";
+
+export const fetcher = (input: RequestInfo, init: RequestInit) =>
+  fetch(input, init).then((res) => res.json());
+
+function createSearchResultList(
+  notes: Omit<Note, "content">[],
+  onNoteSelectChange: (noteId: number) => void
+): JSX.Element | JSX.Element[] {
+
+  return (
+    <ListGroup as="ul" variant="flush">
+      {notes.map((note) => (
+        <ListGroup.Item
+          as="li"
+          key={note.id.toString()}
+          action
+          className="d-flex gap-2 text-nowrap border-0 text-black"
+          onClick={() => onNoteSelectChange(note.id)}
+        >
+          {note.title}
+        </ListGroup.Item>
+      ))}
+    </ListGroup>
+  );
+}
+
+//TODO: search Query will be implemented here or extracted to an external service
+function useSearchResultQuery(searchTerm: String): Omit<Note, "content">[] {
+  //TODO: use searchTerm for future search requests
+  const { data: notes } = useSWR<Omit<Note, "content">[]>(
+    "/api/notes",
+    fetcher
+  );
+
+  var result = notes ?? [];
+
+  if (result.length > 5) {
+    result = result.slice(0, 5);
+  }
+  return result
```

Comment:
You should almost never use `var`, but `const` or `let` instead.
In this case this can also be simplified into
```ts
return notes?.slice(0, 5) ?? []
```
But I am unsure, because that would also mean that if the fetch fails the error goes down. In the future you would also need the mutate function of SWR. So I would design the function like this

```ts
function useSearchResultQuery(searchTerm: string) {
  const { data, error, mutate } = useSWR<Omit<Note, "content">[]>("/api/notes", fetcher);
  return { notes: data?.slice(0, 5), error, mutate };
}
  ```